sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/MagicNumberShouldNotBeUsed.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+
+namespace Tests.Diagnostics
+{
+    public class ValidUseCases
+    {
+        private const int MAGIC = 42;
+        public readonly int MY_VALUE = 25;
+
+        public ValidUseCases()
+        {
+            int i1 = 0;
+            int i2 = -1;
+            int i3 = 1;
+            int i4 = 42;
+
+            for (int i = 0; i < 0; i++)
+            {
+
+            }
+
+            const int VAL = 15;
+
+            Console.Write("test");
+        }
+
+        public override int GetHashCode()
+        {
+            return MY_VALUE * 397;
+        }
+
+        public void Foo(int value = 42)
+        {
+        }
+    }
+
+    public class WrongUseCases
+    {        
+        public WrongUseCases()
+        {
+            Console.WriteLine(12); // Noncompliant
+
+            for (int i = 10; i < 50; i++) // Noncompliant
+            {
+
+            }
+        }
+
+        public int GetValue()
+        {
+            return 13; // Noncompliant
+        }
+    }
+}
+
+#pragma warning disable 1998 //async method lacks an await

```

Comment:
Additional test cases:
* -1 and 1 are not considered magic numbers
* indexers
* array declaration